### PR TITLE
ENH: Compute noise covar rank during inverse

### LINKED ISSUE #224: On missing epochs key
@larsoner I am hitting a corner case where infrequent event key is missing during epoching and causing a `KeyError`. What is your thought on providing a `params.switch` to change the `on_missing` behavior of `Epochs`?

### PR DESCRIPTION
**Pull request by ktavabi**
@larsoner also fixes #224 by adding docstring.

### CODE REVIEW

File: mnefun/_mnefun.py [hunk 1]
```diff
@@ -249,6 +249,13 @@ class Params(Frozen):
     src_pos : float
         Default is 7 mm. Defines source grid spacing for volumetric source
         space.
+    on_missing : string
+        Can set to ‘error’ | ‘warning’ | ‘ignore’. Default is 'error'. Determine
+        what to do if one or several event ids are not found in the recording
+        during epoching. See mne.Epochs docstring for further details.
+    compute_rank : bool
+        Default is False. Set to True to compute rank of the noise covariance
+        matrix during inverse kernel computation.
```
Comment (larsoner) [inline mnefun/_mnefun.py]: The only things that should go here are the arguments of `__init__`. Now would be a good time to add an `Attributes` section to this docstring that includes the things that are available but not in the constructor, starting with these two (can add others later). See:

https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring

Comment (ktavabi) [reply mnefun/_mnefun.py]: LMK if the change is what you are referring to. If so, should I just move all the arguments not part of `__init__` into the attributes section?

Comment (larsoner) [reply mnefun/_mnefun.py]: > should I just move all the arguments not part of __init__ into the attributes section?

Yes if you have time please do

File: mnefun/_mnefun.py [hunk 2]
```diff
@@ -250,6 +250,16 @@ class Params(Frozen):
...
+    Attributes
+    ----------
+    on_missing : string
+        Can set to ‘error’ | ‘warning’ | ‘ignore’. Default is 'error'. Determine
+        what to do if one or several event ids are not found in the recording
+        during epoching. See mne.Epochs docstring for further details.
+    compute_rank : bool
+        Default is False. Set to True to compute rank of the noise covariance
+        matrix during inverse kernel computation.
+
```
Comment (ktavabi) [inline mnefun/_mnefun.py]: Is it necessary to redefine parameters as `self.param = arg` after it's been defined in `__init__`? e.g., `self.ssp_eog_reject = ssp_eog_reject`

Comment (larsoner) [reply mnefun/_mnefun.py]: It is necessary to have this assignment in the code, yes, but it only needs to be documented in the docstring in the Parameters section here.

Comment (ktavabi) [reply mnefun/_mnefun.py]: Why are the assignments necessary?

Comment (larsoner) [reply mnefun/_mnefun.py]: If the line `self.ssp_eog_reject = ssp_eog_reject` were omitted, there would be no `params.ssp_eog_reject`. The instance only gets attributes that you assign to it.
```
>>> class Dummy(object)
...    def __init__(self, param)
...        pass
...
>>> d = Dummy('foo')
>>> d.param
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Dummy' object has no attribute 'param'
```

**Comment by larsoner** [comment]
@mdclarke you might want to start using `compute_rank=True`, it should improve the quality of dSPM for subjects with a lot of head movements

**Comment by larsoner** [comment]
Thanks @ktavabi

**Pull request merged**